qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -334,7 +335,9 @@ def _contains_delay(property_set):
         pm3.append(_swap_check)
         pm3.append(_swap, condition=_swap_condition)
     pm3.append(_unroll)
-    if coupling_map and not coupling_map.is_symmetric:
+    if (coupling_map and not coupling_map.is_symmetric) or (
+        target is not None and target.get_non_global_operation_names(strict_direction=True)
```

Comment:
I _think_ `target.get_non_global_operation_names(strict_direction=True)` is not exactly the analog of `coupling_map.is_symmetric` (but maybe the difference is already being accounted for in the `BasisTranslator`). e.g. If I have two (directional) CX gates, q0->q1 and q1->q2, on my device, `get_non_global_operation_names` would be falsey, but I might still need to correct gate directions.

Reply:
Right it's not 100% the same which is why we still use the `coupling_map.is_symmetric` to trigger adding it to the pass manager here too. We need to do this in addition because if there are any non-global 2q gates in the target we need to correct for direction limitations regardless of whether the coupling map is symmetric. For example, the case in most of the tests with `cx` `(0, 1)` and `ecr` `(1, 0)` the coupling map is symmetric because globally it's `[(0, 1), (1, 0)]` but we need to correct for direction because locally (in terms of each gate) the coupling map is symmetric. This is just basically hacking around doing something like: `any(target.coupling_map(gate).is_symmetric for gate in target if gate is 2q)` (obviously not valid python) which is a more expensive check and not really necessary imo as in the worst case we run these passes a little more than strictly needed. But these direction passes are pretty quick. Like in the worst case you're adding 2 more iterations over the dag so it's not a super big overhead.

Reply:
it's done the layer above in the `transpile()` function. `transpile()` will pull the `CouplingMap` from a `BackendV2` object (which comes from the `target.coupling_map()`) and sets that in the pass manager config. It's still needed for all the layout and routing passes so we can assume it's present. I think in a follow up it might make sense to try and update `transpile()` to use `PassManagerConfig.from_backend` and we can handle this in a single place.

Reply:
oh actually I misread your comment, no it doesn't just passing a target without a backend won't do the right thing and that's a bug. We should fix that I'll push a separate PR.

Reply:
Fixed this in: https://github.com/Qiskit/qiskit-terra/pull/7227/commits/de3984d6483232efbe56f0c0d3fef2afd88838a5 (I also caught another issue around `BackendProperties` while fixing that)